pkg/aiusechat/gemini/gemini-backend.go

```
@@ -0,0 +1,515 @@
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("HTTP request failed: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+
+		// Try to parse as Gemini error
+		var geminiErr GeminiErrorResponse
+		if err := json.Unmarshal(bodyBytes, &geminiErr); err == nil && geminiErr.Error != nil {
+			return nil, nil, nil, fmt.Errorf("Gemini API error (%d): %s", geminiErr.Error.Code, geminiErr.Error.Message)
```

golangci-lint (2.5.0)

[error] ST1005: error strings should not be capitalized

(staticcheck)